### DISCUSSION TITLE
Custom field specific for device (role)

### DISCUSSION THREAD

**Discussion by deku-m** [netbox-community/netbox#13693] [Q&A] [answered]
I have a problem and i dont know how to solve it yet.

But what i want is to create a custom field specific for particular device like for example a firewall.
This custom field only needs to be visible on the device firewall by having a filter device role FWL for example.

What i want to reach is that i have this field for the firewall:
FMC: 1/2/3/4
As we have multiple FMC (Firewall Management Center), i want to add the number on which the firewall is located.
But they only need to be on the firewall and not on all other devices like switches etc.

Is this possible with custom fields?
So that i dont have this custom field on all devices as they will not be visible but hidden.
Anybody have a suggestion/example?

**Comment by candlerb** [comment, marked as answer]
Nope: custom fields either are visible on all objects of a particular type (e.g. DCIM > Device), or not. (There was discussion about how it might be nice to have different custom field visibility per device role, but it's not been accepted as a feature)

However, I can think of a couple of things which might help you.

Firstly, you can make custom fields "Hidden (if unset)":

![image](https://github.com/netbox-community/netbox/assets/44789/a24db6c9-4088-41d0-a852-6e096dee191f)

And secondly, you can use a Custom Validator to stop people setting these fields on devices which don't meet some criteria (e.g. role=Firewall)

**Reply by deku-m** [reply]
Thanks..
Is it also possible to do something like (FMC1)[http://fmc1.server.local] like a a href in a choiceset.

So that i can make a direct click to the underneath combined website?

**Reply by candlerb** [reply]
No, because choicesets are intended for selection via drop-down menus, and the contents of drop-down menus are not links.

You can have a "URL" custom field, but then any URL could be put in there (unless you do your own custom validation)
